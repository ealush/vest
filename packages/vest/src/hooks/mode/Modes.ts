export enum Modes {
  ALL,
  EAGER,
}
