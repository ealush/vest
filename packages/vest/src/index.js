import any from "anyone/any";
import enforce from "n4s/dist/enforce.min";
import { VERSION } from "./constants";
import test from "./core/test";
import validate from "./core/validate";
import { draft, only, skip, warn } from "./hooks";
import { singleton, runWithContext } from "./lib";

export default singleton.register({
  Enforce: enforce.Enforce,
  VERSION,
  any,
  draft,
  enforce,
  only,
  runWithContext,
  skip,
  test,
  validate,
  warn,
});
