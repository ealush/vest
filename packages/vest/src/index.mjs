import enforce from '../../n4s/src/enforce';
import create from './core/suite/create';
import test from './core/test';
import { draft, only, skip, warn, group } from './hooks';

const VERSION = VEST_VERSION;

export default {
  VERSION,
  create,
  draft,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};

export { VERSION, create, draft, enforce, group, only, skip, test, warn };
