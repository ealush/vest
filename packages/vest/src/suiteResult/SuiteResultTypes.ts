import { Maybe } from 'vest-utils';

import { Severity } from 'Severity';
import { SummaryFailure } from 'SummaryFailure';
import { Done } from 'suiteRunResult';
import { SuiteSelectors } from 'suiteSelectors';

export class SummaryBase {
  public errorCount = 0;
  public warnCount = 0;
  public testCount = 0;
  public pendingCount = 0;
}

export class SuiteSummary<
  F extends TFieldName,
  G extends TGroupName
> extends SummaryBase {
  public [Severity.ERRORS]: SummaryFailure<F, G>[] = [];
  public [Severity.WARNINGS]: SummaryFailure<F, G>[] = [];
  public groups: Groups<G, F> = {} as Groups<G, F>;
  public tests: Tests<F> = {} as Tests<F>;
  public valid = false;
}

export type TestsContainer<F extends TFieldName, G extends TGroupName> =
  | Group<G>
  | Tests<F>;
export type GroupTestSummary = SingleTestSummary;

export type Groups<G extends TGroupName, F extends TFieldName> = Record<
  G,
  Group<F>
>;
export type Group<F extends TFieldName> = Record<F, GroupTestSummary>;
export type Tests<F extends TFieldName> = Record<F, SingleTestSummary>;

export type SingleTestSummary = SummaryBase & {
  errors: string[];
  warnings: string[];
  valid: boolean;
  pendingCount: number;
};

export type GetFailuresResponse = FailureMessages | string[];

export type FailureMessages = Record<string, string[]>;

export type SuiteResult<
  F extends TFieldName,
  G extends TGroupName
> = SuiteSummary<F, G> & SuiteSelectors<F, G> & { suiteName: SuiteName };

export type SuiteRunResult<
  F extends TFieldName,
  G extends TGroupName
> = SuiteResult<F, G> & {
  done: Done<F, G>;
};

export type SuiteName = Maybe<string>;

export type TFieldName<T extends string = string> = T;
export type TGroupName<G extends string = string> = G;
