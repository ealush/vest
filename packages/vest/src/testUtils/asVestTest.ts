import { invariant } from 'vest-utils';

import { ErrorStrings } from 'ErrorStrings';
import { TIsolateTest } from 'IsolateTest';
import { isIsolateTest } from 'isIsolateTest';

export function asVestTest(value: any): TIsolateTest {
  invariant(isIsolateTest(value), ErrorStrings.EXPECTED_VEST_TEST);
  return value;
}
