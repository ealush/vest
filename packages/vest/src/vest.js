import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
