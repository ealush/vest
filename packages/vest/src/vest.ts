import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
