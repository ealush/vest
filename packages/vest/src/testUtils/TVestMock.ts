import { TFieldName, TGroupName } from 'SuiteResultTypes';
import * as vest from 'vest';

export type TVestMock = typeof vest;

export type TTestSuiteCallback = (...args: any[]) => void;
export type TTestSuite = vest.Suite<TFieldName, TGroupName, TTestSuiteCallback>;
