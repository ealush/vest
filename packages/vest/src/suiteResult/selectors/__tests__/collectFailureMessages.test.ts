import { TTestSuite } from 'testUtils/TVestMock';

import { dummyTest } from '../../../testUtils/testDummy';

import { Modes } from 'Modes';
import { SuiteResult } from 'SuiteResultTypes';
import { create, group } from 'vest';
import * as vest from 'vest';

describe('collectFailureMessages', () => {
  let suite: TTestSuite;
  let res: SuiteResult<string, string>;

  test('Result has an array of matching error messages', () => {
    const result = res.getErrors();
    expect(result.field_1).toEqual([
      'field_1_failure message 2',
      'field_1_failure message 3',
    ]);
  });

  it('Should return filtered messages by the selected group', () => {
    const result = res.getErrorsByGroup('group_1');

    expect(result).toEqual({
      field_1: [],
      field_2: ['field_2_failure message 1', 'field_2_failure message 3'],
    });
  });

  it('Should return an empty object when no options and no failures', () => {
    expect(res.getErrors('nonexistent_field')).toEqual([]);
  });

  it('Should return an object with an empty array when selected field has no errors', () => {
    expect(res.getErrors('v')).toEqual([]);
  });

  describe('getErrors', () => {
    describe('When no options passed', () => {
      it('should match snapshot', () => {
        expect(res.getErrors()).toMatchSnapshot();
      });
    });

    describe('When specific field requested', () => {
      it('Should match snapshot', () => {
        expect(res.getErrors('field_1')).toMatchSnapshot();
        expect(res.getErrors('field_2')).toMatchSnapshot();
        expect(res.getErrors('field_3')).toMatchSnapshot();
      });
    });
  });

  describe('getWarnings', () => {
    describe('When no options passed', () => {
      it('should match snapshot', () => {
        expect(res.getWarnings()).toMatchSnapshot();
      });
    });

    describe('When specific field requested', () => {
      it('Should match snapshot', () => {
        expect(res.getWarnings('field_1')).toMatchSnapshot();
        expect(res.getWarnings('field_2')).toMatchSnapshot();
        expect(res.getWarnings('field_3')).toMatchSnapshot();
      });
    });
  });

  describe('getErrorsByGroup', () => {
    it('Should match snapshot', () => {
      expect(res.getErrorsByGroup('group_1')).toMatchSnapshot();
    });

    describe('with field name', () => {
      it('Should match snapshot', () => {
        expect(res.getErrorsByGroup('group_1', 'field_1')).toMatchSnapshot();
        expect(res.getErrorsByGroup('group_1', 'field_2')).toMatchSnapshot();
        expect(res.getErrorsByGroup('group_1', 'field_3')).toMatchSnapshot();
      });
    });
  });
  describe('getWarningsByGroup', () => {
    it('Should match snapshot', () => {
      expect(res.getWarningsByGroup('group_1')).toMatchSnapshot();
    });

    describe('with field name', () => {
      it('Should match snapshot', () => {
        expect(res.getWarningsByGroup('group_1', 'field_1')).toMatchSnapshot();
        expect(res.getWarningsByGroup('group_1', 'field_2')).toMatchSnapshot();
        expect(res.getWarningsByGroup('group_1', 'field_3')).toMatchSnapshot();
      });
    });
  });

  beforeEach(() => {
    suite = create(() => {
      vest.mode(Modes.ALL);
      dummyTest.passing('field_1', 'field_1_failure message 1');
      dummyTest.failing('field_1', 'field_1_failure message 2');
      dummyTest.failing('field_1', 'field_1_failure message 3');
      group('group_1', () => {
        vest.test('field_1', () => false);
        vest.test('field_3', () => {});
        dummyTest.failing('field_2', 'field_2_failure message 1');
        dummyTest.failingWarning('field_2', 'field_2_warning message 3');
        dummyTest.failing('field_2', 'field_2_failure message 3');
      });
      dummyTest.passingWarning('field_2', 'field_2_warning message 1');
      dummyTest.failingWarning('field_2', 'field_2_warning message 2');
      dummyTest.passing('v');

      group('group_2', () => {
        dummyTest.passing('x');
      });
    });
    suite();
    res = suite.get();
  });
});
