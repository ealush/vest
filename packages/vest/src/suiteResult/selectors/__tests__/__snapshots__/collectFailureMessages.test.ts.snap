// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectFailureMessages getErrors When no options passed should match snapshot 1`] = `
{
  "field_1": [
    "field_1_failure message 2",
    "field_1_failure message 3",
  ],
  "field_2": [
    "field_2_failure message 1",
    "field_2_failure message 3",
  ],
}
`;

exports[`collectFailureMessages getErrors When specific field requested Should match snapshot 1`] = `
[
  "field_1_failure message 2",
  "field_1_failure message 3",
]
`;

exports[`collectFailureMessages getErrors When specific field requested Should match snapshot 2`] = `
[
  "field_2_failure message 1",
  "field_2_failure message 3",
]
`;

exports[`collectFailureMessages getErrors When specific field requested Should match snapshot 3`] = `[]`;

exports[`collectFailureMessages getErrorsByGroup Should match snapshot 1`] = `
{
  "field_1": [],
  "field_2": [
    "field_2_failure message 1",
    "field_2_failure message 3",
  ],
}
`;

exports[`collectFailureMessages getErrorsByGroup with field name Should match snapshot 1`] = `[]`;

exports[`collectFailureMessages getErrorsByGroup with field name Should match snapshot 2`] = `
[
  "field_2_failure message 1",
  "field_2_failure message 3",
]
`;

exports[`collectFailureMessages getErrorsByGroup with field name Should match snapshot 3`] = `[]`;

exports[`collectFailureMessages getWarnings When no options passed should match snapshot 1`] = `
{
  "field_2": [
    "field_2_warning message 3",
    "field_2_warning message 2",
  ],
}
`;

exports[`collectFailureMessages getWarnings When specific field requested Should match snapshot 1`] = `[]`;

exports[`collectFailureMessages getWarnings When specific field requested Should match snapshot 2`] = `
[
  "field_2_warning message 3",
  "field_2_warning message 2",
]
`;

exports[`collectFailureMessages getWarnings When specific field requested Should match snapshot 3`] = `[]`;

exports[`collectFailureMessages getWarningsByGroup Should match snapshot 1`] = `
{
  "field_2": [
    "field_2_warning message 3",
  ],
}
`;

exports[`collectFailureMessages getWarningsByGroup with field name Should match snapshot 1`] = `[]`;

exports[`collectFailureMessages getWarningsByGroup with field name Should match snapshot 2`] = `
[
  "field_2_warning message 3",
]
`;

exports[`collectFailureMessages getWarningsByGroup with field name Should match snapshot 3`] = `[]`;
